## Survial Rate of Passanger in titanic

In [23]:
import pandas as pd
df = pd.read_csv('titanic.csv')
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

#### Droping the columns that we don't need 

In [24]:
df.drop(['PassengerId','Name','SibSp','Parch','Ticket','Cabin','Embarked'], axis = 'columns', inplace = True)

In [25]:
df.head()

Survived  Pclass     Sex   Age     Fare
0         0       3    male  22.0   7.2500
1         1       1  female  38.0  71.2833
2         1       3  female  26.0   7.9250
3         1       1  female  35.0  53.1000
4         0       3    male  35.0   8.0500

#### Targeting the survial rate of Pasanger

In [26]:
inputs = df.drop('Survived', axis= 'columns')
target = df.Survived

In [40]:
inputs.head()

Pclass     Sex   Age     Fare  Sex_n
0       3    male  22.0   7.2500      1
1       1  female  38.0  71.2833      0
2       3  female  26.0   7.9250      0
3       1  female  35.0  53.1000      0
4       3    male  35.0   8.0500      1

In [41]:
inputs.Age = inputs.Age.fillna(inputs.Age.mean())

Labling the sex column

In [42]:
from sklearn.preprocessing import LabelEncoder

In [43]:
le_sex = LabelEncoder()

In [47]:
inputs['Sex_n'] = le_sex.fit_transform(inputs['Sex'])
inputs_n = inputs.drop(['Sex'], axis = 'columns')

In [48]:
inputs_n.head()

Pclass   Age     Fare  Sex_n
0       3  22.0   7.2500      1
1       1  38.0  71.2833      0
2       3  26.0   7.9250      0
3       1  35.0  53.1000      0
4       3  35.0   8.0500      1

In [51]:
from sklearn import tree
model = tree.DecisionTreeClassifier()

In [52]:
model.fit(inputs_n, target)

DecisionTreeClassifier(class_weight=None, criterion='gini', max_depth=None,
                       max_features=None, max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, presort=False,
                       random_state=None, splitter='best')

Let's predict the Death Rate of passanger whose id is pclass is 1, age 35, Fare 53.100, and sex is female.

In [54]:
model.predict([['1','35','53.1','0']])

array([1], dtype=int64)

score of model

In [55]:
model.score(inputs_n,target)

0.9797979797979798